src/epoch/gregorian.rs

```diff
@@ -269,45 +251,43 @@ impl Epoch {
                     source: DurationError::Underflow,
                 })
             }
-            Some(years_since_ref) => match years_since_ref.checked_mul(365) {
+            Some(years_since_ref) => match years_since_ref.checked_mul(DAYS_PER_YEAR_NLD as i32) {
                 None => {
                     return Err(EpochError::Duration {
                         source: DurationError::Overflow,
                     })
                 }
                 Some(days) => Unit::Day * i64::from(days),
             },
-        } - time_scale.gregorian_epoch_offset();
```

Comment:
Moved to the end of the function to make it easier to write the reverse algorithm.